### ISSUE TITLE
Bug: Division between different types does not show an error

### ISSUE THREAD

**Issue by gptsarthak** [lcompilers/lpython#1685]
Division between any combination of ```i32,i64,f32,f64``` and ```bool``` does not show any type mismatch error and returns a ```f64``` value.
```Python
def main():
    a : i32 = 10
    b : f64 = 20.0
    c : f64 = a / b
    d : f64 = b / a
    print(c)          # 0.5
    print(d)          # 2.0

    e : bool = True
    f : i32 = 15
    g : f64 = e / f
    h : f64 = f / e
    print(g)          # 0.066
    print(h)          # 15.0

main()
```
```console
(base) sarthak@pop-os:~/lpython$ python examples/div.py
0.5
2.0
0.06666666666666667
15.0
(base) sarthak@pop-os:~/lpython$ lpython examples/div.py
5.00000000000000000e-01
2.00000000000000000e+00
6.66666666666666657e-02
1.50000000000000000e+01
```
Though it has the same output as CPython, I don't think this is the intended behavior for LPython. If we do not support multiplication between these types, we should not support division either.

**Comment by certik** [comment]
Yes, these should raise an error message and require an explicit casting. Thanks for reporting it!
